### PR TITLE
fix(content-block-simple): storybook cleanup

### PR DESCRIPTION
**Pull request by NSDrowned**
### Related Ticket(s)

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/5527

This PR also fixes all issues in https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/5883

### Description

Storybook fixes for content block simple

### CODE REVIEW

**Comment by kennylam** [comment]
@NSDrowned is this a duplicate of #5948?

**Comment by NSDrowned** [comment]
@kennylam almost... This issue required some of the fixes I did on the other issue to work. For example, the complementary border needed the knobs I fixed on the other one. This happened because this was originally 1 issue that was split into multiple issues after I had already started. I'm going to close first one and mention it in this PR since this one includes all the fixes.

File: packages/web-components/src/components/content-block-simple/__stories__/content-block-simple.stories.scss [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  dds-link-list {
+    /* stylelint-disable-next-line selector-type-no-unknown */
+    dds-link-list-item-card-cta {
+      /* stylelint-disable-next-line selector-type-no-unknown */
+      dds-card-cta-footer {
+        margin-top: 0;
+        display: flex;
+        align-items: center;
+      }
+    }
```
Comment (kennylam) [inline packages/web-components/src/components/content-block-simple/__stories__/content-block-simple.stories.scss:23]: @NSDrowned Sorry I missed this in the last review. What happens when this component is used outside Storybook and these styles aren't there? Also, I think these styles are misaligning the icons?

Comment (NSDrowned) [reply packages/web-components/src/components/content-block-simple/__stories__/content-block-simple.stories.scss:23]: @kennylam np... the problem only occurs in this story. 

From my tests they looked perfect outside:

image

I'll take another look at why it's happening.

Comment (kennylam) [reply packages/web-components/src/components/content-block-simple/__stories__/content-block-simple.stories.scss:23]: @NSDrowned Oh hm, yes you're right I can see what you mean. Looks like you found a bug and these styles should be added to the codebase 👍

**Comment by DragosRistici** [comment]
Hey, Julio, thank you for taking this on. A few little things:
<br>
### 1. Spacing after the heading should be 32px like in the specs. Right now its 48px.
Screenshot 2021-05-12 at 16 01 21
Screenshot 2021-05-12 at 15 40 58
<br>

### 2. Web Components - on pressed, the CTA card lacks the `focus` state
<br><br>
### 3. React story with link list has the same heading grid error
Screenshot 2021-05-12 at 16 28 18Screenshot 2021-05-12 at 16 43 10

### 4. The React Wrapper page is kinda off the grid. See below. Let me know if I can help in some way.
Screenshot 2021-05-12 at 16 44 50
Screenshot 2021-05-12 at 16 44 43

Screenshot 2021-05-12 at 14 09 34Screenshot 2021-05-12 at 14 09 26Screenshot 2021-05-12 at 14 09 21


<br><br> Let me know if I can help in any way. Thanks!

**Comment by DragosRistici** [comment]
Thank you, @NSDrowned. So the spacing below heading will be solved with [5227](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/5227) and the link list grid alignment [issue](https://ibmdotcom-react-canary.mybluemix.net/?path=/story/components-content-block-simple--with-aside-elements) will take care of it here, too.

**Review by DragosRistici — APPROVED** [review]

**Review by ariellalgilmore — APPROVED** [review]
LGTM!

**Pull request merged**